caluma/core/ordering.py

```diff
@@ -0,0 +1,159 @@
+from typing import Any, Tuple, Union
+
+from django import forms
+from django.db.models import OuterRef, Subquery
+from django.db.models.expressions import CombinedExpression, F, Value
+from django.db.models.query import QuerySet
+from django_filters.fields import ChoiceField
+from django_filters.rest_framework import Filter
+from graphene import Enum
+from graphene_django.forms.converter import convert_form_field
+from graphene_django.registry import get_global_registry
+from rest_framework import exceptions
+
+from caluma.form.models import Answer, Question
+
+OrderingFieldType = Union[F, str]
+
+
+class CalumaOrdering(Filter):
+    """Base class for new-style ordering filters.
+
+    This is not really a filter in the rest framework sense,
+    but we use it's infrastructure to get conversion to GraphQL
+    types.
+
+    This is an abstract base class. To use it, you need to
+    implement the `get_ordering_value()` method to return a tuple
+    of `(queryset, ordering_field)`. The queryset may be modified,
+    for example if you need to annotate it to get to the desired
+    ordering field.
+    """
+
+    def get_ordering_value(
+        self, qs: QuerySet, value: Any
+    ) -> Tuple[QuerySet, OrderingFieldType]:  # pragma: no cover
+        raise NotImplementedError("get_ordering_value() needs to be implemented")
+
+
+class MetaFieldOrdering(CalumaOrdering):
+    """Ordering filter for ordering by `meta` values."""
+
+    def __init__(self, field_name="meta"):
+        super().__init__()
+        self.field_name = field_name
+
+    field_class = forms.CharField
+
+    def get_ordering_value(
+        self, qs: QuerySet, value: Any
+    ) -> Tuple[QuerySet, OrderingFieldType]:
+
+        return qs, CombinedExpression(F(self.field_name), "->", Value(value))
+
+
+class AnswerValueOrdering(CalumaOrdering):
```

Comment:
This should be in the `form` app, as `core` should never depend on `form`.

Reply:
Right you are! Fixed